src/components/ShopList.vue

```diff
@@ -0,0 +1,55 @@
+<template>
+    <div class="shops-list">
+        <article class="shop" v-for="shop in shops">
+           <img :src="shop.img" class="shop__img" />
+           <span class="shop__name">{{ shop.name }}</span>
+           <span class="shop__distance">{{ shop.distance }}</span>
+           <address class="shop__address">{{ shop.street }}</address>
+           <router-link :to='{path: `/shop-on-map/${shop.id}`}'>View on map</router-link>
+       </article>
+    </div>
+</template>
+
+<script>
+
+export default
+{
+    name: 'shop-list',
+    props: ['shops'],
```

Comment:
All data that are from the server should go by the same river (trough vuex store). 
take to account this example:
You are getting data from server to set them for render. What if other component is depending on them but isn't a child of SearchShopByName.vue component. Receiving by props isn't possible. So in this case We can leave this for this components parent -> children relation but shop list should be available globally for the app trough vuex store